quarkus/tests/integration/src/main/java/org/keycloak/it/junit5/extension/CLITestExtension.java

```diff
@@ -100,7 +103,34 @@ public void beforeEach(ExtensionContext context) throws Exception {
             onBeforeStartDistribution(context.getRequiredTestMethod().getAnnotation(BeforeStartDistribution.class));
 
             if (launch != null) {
-                result = dist.run(Arrays.asList(launch.value()));
+                List<String> argsFromLaunchAnnotation = new ArrayList<>(Arrays.asList(launch.value()));
```

Comment:
Is it possible to separate this logic from the BeforeEach method? It's getting started to be really messy.

Reply:
Done! It was created a private method to handle this behavior.